🌱 Update instructions checklist to generate release notes (#9443)
kubernetes-sigs/cluster-api: docs/release/release-tasks.md

   <br>Prior art: [Add release 1.2 book link](https://github.com/kubernetes-sigs/cluster-api/pull/6697)

#### Create PR for release notes
1. Checkout the `main` branch.
1. Build the release note tools binary.
   ```bash
   go build -o bin/notes hack/tools/release/notes.go

>    ```

Reviewer: ❓ : `make release-notes` has special handling about e.g. `PRE_RELEASE` to add some text as header:

https://github.com/kubernetes-sigs/cluster-api/blob/1735aa5550db514a0dd00b19a1d307e07766b885/Makefile#L1086-L1091

 I may not be up to date on the tooling but shouldn't we be able to do sth like the following (and by that be able to skip the step of compiling the notes binary?)
  
 ```sh
 RELEASE_NOTES_DIR=CHANGELOG
 PREVIOUS_TAG=v1.3.x
 PRE_RELEASE=y
 RELEASE_TAG=v1.4.x
 make release-notes
 ```
 
 This way we would make sure (after checking out main) that we always use the latest version of the tool and not be able to be lazy and skip building the new version.

Author: The problem with this is the tool needs to be run on the branch you are releasing from. If we want to do this, we need to backport all release notes tool improvements to the release branches so they are available when running the target. Compiling to binary from main allows to avoid this process.

We have an improvement task to be able to run the tool from any branch by reading directly  from github instead of the local git. But we haven't got to it yet.

1. Checkout the latest commit on the release branch, e.g. `release-1.4`, or the main branch if the release branch doesn't yet exist (e.g. beta release).
1. Generate release notes with:

   ```bash
   # PREVIOUS_TAG should be the last patch release of the previous minor release.
   PREVIOUS_TAG=v1.3.x
   # RELEASE_TAG should be the new desired tag (note: at this point the tag does not yet exist).
   RELEASE_TAG=v1.4.x

>    # If this is a beta or RC release, add the --pre-release-version flag

Reviewer 2: do you need to set `--pre-release-version` flag, i.e `--pre-release-version=v1.6.0-beta.0`?

Author: `--pre-release-version` is just a boolean flag
It enables/disables the "pre release" block text, but it doesn't take any value as argument

   ./bin/notes --from=$PREVIOUS_TAG > CHANGELOG/${RELEASE_TAG}.md
   ```

1. This will generate a new release notes file at `CHANGELOG/<RELEASE_TAG>.md`. Finalize the release notes:
    - [ ] Update the `Kubernetes version support section`. If this is a patch release you can most probably copy the same values from the previous patch release notes. Except if this is the release where a new Kubernetes version support is added.
       <br>**Note**: Check our [Kubernetes support policy](https://cluster-api.sigs.k8s.io/reference/versions.html#supported-kubernetes-versions) in the CAPI book. In case of doubt, reach out to the current release lead.

>     - [ ] If this is a `vX.X.0` release, add fill the content for the `Highlights` section. Otherwise, remove the section all together.

Reviewer 2: ```suggestion
    - [ ] If this is a `vX.X.0` release, fill in the content for the `Highlights` section. Otherwise, remove the section altogether.
```

>     - [ ] If there a deprecations in this release (for example, a CAPI API version drop), add them, to the `Deprecation Warning` section. Otherwise, remove the section all together.

Reviewer 2: ```suggestion
    - [ ] If there a deprecations in this release (for example, a CAPI API version drop), add them, to the `Deprecation Warning` section. Otherwise, remove the section altogether.
```

    - [ ] Look for any `MISSING_AREA` entries. Add the corresponding label to the PR and regenerate the notes.

>     - [ ] Look for any `MULTIPLE_AREAS` entry. If the PR does indeed guarantee multiple areas, just remove the `MULTIPLE_AREAS` prefix and just leave the areas. Otherwise, fix the labels in the PR and regenerate the notes.

Reviewer 2: to follow the previous line 
```suggestion
    - [ ] Look for any `MULTIPLE_AREAS` entries. If the PR does indeed guarantee multiple areas, just remove the `MULTIPLE_AREAS` prefix and just leave the areas. Otherwise, fix the labels in the PR and regenerate the notes.
```

    - [ ] Review that all areas are correctly assigned to each PR. If not, correct the labels and regenerate the notes.
    - [ ] Look for area duplications in PR title. Sometimes authors add a prefix in their PR title that matches the area label. When the notes are generated, the area is as a prefix to the PR title, which can create redundant information. Remove the one from the PR title and just leave the area. Make sure you capitalize the title after this.
    - [ ] Check that all entries are in the right section. Sometimes the wrong emoji prefix is added to the PR title, which drives the section in which the entry is added in the release notes. Manually move any entry as needed. Note that fixing the PR title won't fix this even after regenerating the notes, since the notes tool reads this info from the commit messages and these don't get rewritten.
    - [ ] Sort manually all entries if you made any manual edits that might have altered the correct order.

>     - [ ] **For minor releases** Modify `Changes since v1.x.y` to `Changes since v1.x`

Reviewer 2: ```suggestion
    - [ ] **For minor releases:** Modify `Changes since v1.x.y` to `Changes since v1.x`
```
